### PR TITLE
Fix CMake

### PR DESCRIPTION
**Pull request by lrapetti**
This PR aims to fix the CMake configuration of HDE with two improvements:
- remove dependency from [robotology-playground/xsens-mvn](https://github.com/robotology-playground/xsens-mvn) (It will no longer be necessary with `HDEv2` that will depend upon [wearables](https://github.com/robotology-playground/wearables)) https://github.com/robotology/human-dynamics-estimation/commit/914332f7bd99a1b68778f79921679a73194acf4c
- fix the CMake configuration for HumanROSMsgs since it is a header-only library (with the current configuration, it is preventing HDE to be installed on `Windows`) https://github.com/robotology/human-dynamics-estimation/commit/d716e44ac7c604f35178900299de401d7f7da228
- Force plugins installation as Dynamic Libraries https://github.com/robotology/human-dynamics-estimation/pull/87/commits/d84cd1f24e03ea48c6ed031b72a177d865247a19

### CODE REVIEW

**Comment by lrapetti** [comment]
@diegoferigo Can you review the changes?

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,9 +44,11 @@ if(MSVC)
 else()
     option(BUILD_SHARED_LIBS "Build libraries as shared as opposed to static" ON)
 endif()
-if (NOT BUILD_SHARED_LIBS)
-    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-endif()
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (traversaro) [inline CMakeLists.txt]: What is the rationale for this change?

Comment (lrapetti) [reply CMakeLists.txt]: > What is the rationale for this change?

So this was a trap?!

Comment (traversaro) [reply CMakeLists.txt]: > So this was a trap?!

I simply forgot about what our recommended  way of handling `CMAKE_POSITION_INDEPENDENT_CODE` was, and I had to check `how-to-export-cpp-library` to remember the details.

**Comment by diegoferigo** [comment]
It is not compiling on Ubuntu 16.04. I think it is related to the compiler that is too old, iDynTree does not support it. We should have a discussion to define minimum requirements for the AnDy related software. Since there are no users beyond us and considering that it is a fairly new project (especially wearables), I would tend to be as less conservative as possible.

**Comment by traversaro** [comment]
> iDynTree does not support it

I think iDynTree should have full support for Ubuntu 16.04, no? 

> Since there are no users beyond us and considering that it is a fairly new project (especially wearables), I would tend to be as less conservative as possible.

Even if the project is new, as it is strictly related to Andy project, I think you should evaluate (depending on the future project demo), if it may make sense to consider if we need to be compatible with some environments of the project partners. A typical case with robotics academic lab is that sometimes they are bound to relatively old Ubuntu versions (not the latest LTS, for example) due to the need to remain compatible with some specific LTS ROS release. @DanielePucci

**Comment by diegoferigo** [comment]
> I think iDynTree should have full support for Ubuntu 16.04, no?

Are you sure that the default compiler version on 16.04 is compatible with iDynTree?

> I think you should evaluate (depending on the future project demo), if it may make sense to consider if we need to be compatible with some environments of the project partners.

We should have a meeting on this

**Comment by traversaro** [comment]
> Are you sure that the default compiler version on 16.04 is compatible with iDynTree?

At least a subset of the possible configuration combinations are tested in Travis on 16.04 : https://travis-ci.org/robotology/idyntree .

**Comment by diegoferigo** [comment]
I probably got confused with https://github.com/robotology/idyntree/pull/436. Maybe the incompatibility was before switching to docker when Travis was not yet using Ubuntu Xenial.

**Comment by traversaro** [comment]
Just to be clear, I am not opposing at all dropping support for 16.04, I just suggest to double check with @DanielePucci .

**Comment by diegoferigo** [comment]
@lrapetti  `HumanDynamicsEstimator` does not compile on ubuntu xenial, check the Travis build of this PR

**Comment by DanielePucci** [comment]
@diegoferigo @lrapetti 
> We should have a meeting on this

Feel free to discuss this whenever you want if compilation on 16.04 will not be fixed

**Comment by lrapetti** [comment]
This PR is deprecated after https://github.com/robotology/human-dynamics-estimation/pull/100.
closing

**Comment by traversaro** [comment]
I think we can close this PR, as after https://github.com/robotology/human-dynamics-estimation/pull/100 it outdated.

**Comment by traversaro** [comment]
@lrapetti
